src/EnergyPlus/PlantLoopHeatPumpEIR.cc

```diff
@@ -578,7 +753,37 @@ void EIRPlantLoopHeatPump::capModFTCurveCheck(EnergyPlusData &state, const Real6
     }
 }
 
-void EIRPlantLoopHeatPump::eirModCurveCheck(EnergyPlusData &state, Real64 &eirModifierFuncTemp, Real64 &eirModifierFuncPLR)
+void EIRPlantLoopHeatPump::heatRecoveryCapModFTCurveCheck(EnergyPlusData &state,
+                                                          const Real64 loadSideOutletSetpointTemp,
+                                                          Real64 &capacityModifierFuncTemp)
+{
+    if (capacityModifierFuncTemp < 0.0) {
+        if (this->heatRecCapModFTErrorIndex == 0) {
+            ShowSevereMessage(state, format("{} \"{}\":", DataPlant::PlantEquipTypeNames[static_cast<int>(this->EIRHPType)], this->name));
+            ShowContinueError(state,
+                              format(" Heat Recovery mode Capacity Modifier curve (function of Temperatures) output is negative ({:.3T}).",
+                                     capacityModifierFuncTemp));
+            ShowContinueError(
+                state,
+                format(
+                    " Negative value occurs using a load side water temperature of {:.2T}C and heat recovery entering water temperature of {:.2T}C.",
+                    loadSideOutletSetpointTemp,
+                    this->heatRecoveryInletTemp));
+            ShowContinueErrorTimeStamp(state, " Resetting curve output to zero and continuing simulation.");
+        }
+        ShowRecurringWarningErrorAtEnd(
+            state,
+            format("{} \"{}\": Heat Recovery mode Capacity Modifier curve (function of Temperatures) output is negative warning continues...",
+                   DataPlant::PlantEquipTypeNames[static_cast<int>(this->EIRHPType)],
+                   this->name),
+            this->heatRecCapModFTErrorIndex,
+            capacityModifierFuncTemp,
+            capacityModifierFuncTemp);
+        capacityModifierFuncTemp = 0.0;
+    }
+}
```

Comment:
Heat recovery capacity modifier curve value check function. Resets the value to `0` if negative.